File: tested/languages/python/templates/evaluation_utils.py
from dataclasses import dataclass, field
from typing import List, Optional


@dataclass
class Message:
    description: str
    format: str = "text"
    permission: Optional[str] = None


@dataclass
class EvaluationResult:
    result: bool
    readable_expected: Optional[str] = None
    readable_actual: Optional[str] = None
    messages: List[Message] = field(default_factory=list)
    dsl_expected: Optional[str] = None
    dsl_actual: Optional[str] = None

File: tested/languages/haskell/templates/EvaluationUtils.hs
{-# LANGUAGE OverloadedStrings #-}
module EvaluationUtils where

data Message = Message {
    description :: String,
    format :: String,
    permission :: Maybe String
} deriving Show

data EvaluationResult = EvaluationResult {
    result :: Bool,
    readableExpected :: Maybe (String),
    readableActual :: Maybe (String),
    messages :: [Message],
    dslExpected :: Maybe (String),
    dslActual :: Maybe (String)
} deriving Show

message description = Message {
    description = description,
    format = "text",
    permission = Nothing
}

evaluationResult = EvaluationResult {
    result = False,
    readableExpected = Nothing,
    readableActual = Nothing,
    dslExpected = Nothing,
    dslActual = Nothing,
    messages = []
}
